File: frontend/api_agent/mira/prompt_builder.py
def build_prompt(language_option, recipe_name, model):
      
   prompt_template = f"""
  You are an AI agent for the purpose of temporal reasoning. Your name is MIRA AI agent. You have to follow the protocols listed in MIRA PROTOCOL. I'll give you a list of available Objects (OBJ) and Valid Actions. Pickout correct objects M¯im¯ams¯a Inspired Framework- Temporal Reasonin from the list of Available Objects. Each action duration (DUR) is also given in valid action. The starting condition (START) and the end goal (FINAL GOAL) for the sequence is also given. With these you have to generate a sequence of instructions for the given task description. For each instruction also provide the type (TYPE) of instruction from MIRA PROTOCOL

   Recipe Name: {recipe_name}

   TYPE: NON INSTRUCTION 
   • Example: Brown ground meat and onion in a large pot
   TYPE: SIMPLE INSTRUCTION
   • Instructions can include one object or multiple objects (e.g. take pen or take pens and papers).
   • Example: Add beef broth and celery
   TYPE: SEQUENTIAL INSTRUCTION
   • If two actions are provided in a sequence, use “then” and do not use “and” (e.g. Take pen then write).
   • Example: Cover the pot then simmer for 20 minutes
   TYPE: INSTRUCTION WITH REASON
   • The instruction includes an action followed by a reason or condition.
   • Example: Add a little salt to taste during the last hour of cooking
   TYPE: INSTRUCTION WITH PURPOSE
   • Each instruction can include a goal—here the instruction should have the intention of goal perspective (e.g. Take pen if you want to write).
   • Example: Let it stand in a cool place if your intention is to make it firm
   TYPE: EXCLUSIVE INSTRUCTION
   • Each instruction can include multiple objects which are exclusive (e.g. take pen or pencil).
   • If there are options between actions, include "or" within that instruction.
   • Example: Heat the oil or margarine in a soup pot
   TYPE: INSTRUCTION WITH RESTRICTION
   • These instructions have a condition or limit applied to how the action must be performed.
   • Example: Add the reserved liquid only until the desired consistency is reached
   TYPE: MANDATORY INSTRUCTION
   • Instructions that must be followed strictly. The action is compulsory.
   • Example: You must add the onion and garlic
   TYPE: PARALLEL INSTRUCTION
   • Two or more actions or objects are handled simultaneously.
   • Example: Cream the sugar and the butter simultaneously until light and whipped
   TYPE: INSTRUCTION WITH REASON AND PURPOSE
   • The instruction includes both a reason and an intended outcome.
   • Example: Add more soy sauce if needed, to enhance the taste
   TYPE: INSTRUCTION WITH REASON WITH RESTRICTION
   • Action is performed under a restriction, and also includes a reason or condition.
   • Example: Working in batches, puree in a blender until thick and smooth
   TYPE: INSTRUCTION WITH SEQUENCE WITH RESTRICTION
   • Two actions are done in a sequence, with some restriction/condition on how it is done.
   • Example: Dip chicken pieces into soup mixture then turn only to coat all over

   give the result in this format 
   1. Instruction
   <span style="color:green"> Type: Simple Instruction </span> (Its classification)
   ...


   Available Objects = {{Slices of bread White bread, Whole wheat bread, Multi-grain
   bread, Condiments, Mayonnaise, Mustard, Butter, Ketchup, Spreads, Vegetables,
   Meat, Cheese, Sandwich maker, Grill, Water, Teapot 
   
   Valid Actions = {{ Take OBJ: DUR 1 minute, Heat OBJ: DUR 3 minutes, Pickup

   Now, generate the step-by-step instructions for {recipe_name} following the above structured format.
      """

   return prompt_template

File: frontend/customFlow/prompt_builder.js
export const buildPrompt = (taskName) => {
	return `
You are an AI agent for the purpose of temporal reasoning. Your name is MIRA AI agent. You have to follow the protocols listed in MIRA PROTOCOL. I'll give you a list of available Objects (OBJ) and Valid Actions. Pick out correct objects from the list of Available Objects. Each action duration (DUR) is also given in valid action. The starting condition (START) and the end goal (FINAL GOAL) for the sequence is also given. With these you have to generate a sequence of instructions for the given task description. For each instruction also provide the type (TYPE) of instruction from MIRA PROTOCOL.

Task: ${taskName}

**MIRA Protocol:**  
- Each instruction must specify action and object(s).
- For each instruction, state the required and resulting object states.
- Classify each instruction as: Simple Instruction, Instruction with Goal, Instruction in Sequence, Exclusive Instruction, or Mandatory Instruction as per the following rules
**RULES**
INSTRUCTION TYPES:
 1. SIMPLE INSTRUCTION:
    - Generate one action per instruction.
    - Instructions can include one or more objects.
    - Examples:
      • Take pens → TYPE: SIMPLE INSTRUCTION
      • Take pens using hand → TYPE: SIMPLE INSTRUCTION

    2. INSTRUCTION WITH PURPOSE:
    - Include a goal or intention behind the action.
    - Examples:
      • Take pen if you want to write
      • Take pen if you have the intention of writing
      → TYPE: INSTRUCTION WITH PURPOSE

    3. EXCLUSIVE INSTRUCTION (OBJECTS):
    - Multiple objects given, only one to be chosen.
    - Example:
      • Take pen or pencil
      → TYPE: EXCLUSIVE INSTRUCTION

    4. EXCLUSIVE INSTRUCTION (ACTIONS):
    - Use ‘or’ between alternative actions.
    - Example:
      • Go by walk or take a car to reach destination
      → TYPE: EXCLUSIVE INSTRUCTION

    5. INSTRUCTION WITH SEQUENCE:
    - If two actions must be performed in order or sequence, use “then”.
    - Do NOT use “and”.
    - Example:
      • Take pen then write
      . Go left then right
      . Eat food then drink water
     
      → TYPE: INSTRUCTION WITH SEQUENCE

    6. MANDATORY INSTRUCTION:
    - Both actions must be performed, order does not matter.
    - Examples:
      • Take pen and paper 
      • Write test and be calm 
      . Add noodles and pick up tomato
        → TYPE: MANDATORY INSTRUCTION

Available Objects = {Slices of bread, White bread, Whole wheat bread, Multi-grain bread, Condiments, Mayonnaise, Mustard, Butter, Ketchup, Spreads, Vegetables, Meat, Cheese, Sandwich maker, Grill, Water, Teapot, Stove, Knife, Cutting board, Cup, Tea bags, Sugar, Milk, Spoon, Coffee beans, Coffee maker, Filter, Mug, Kettle, Cream}

Valid Actions = {Take OBJ: DUR 1 minute, Heat OBJ: DUR 3 minutes, Pickup OBJ: DUR 1 minute, Cut OBJ: DUR 2 minutes, Mix OBJ: DUR 2 minutes, Pour OBJ: DUR 1 minute, Add OBJ: DUR 1 minute, Wait: DUR variable, Serve OBJ: DUR 1 minute}

Please generate the step-by-step instructions for ${taskName} and return the result as a JSON array with the following format:
[
  {
    "instruction": "Step description here",
    "type": "Instruction type from the list above"
  }
]
**Output your answer in the format shown above.**

Only return valid JSON, no additional text or formatting.
`;
};
